python/hpsmc/alignment/_pede.py

```diff
@@ -0,0 +1,163 @@
+"""! running pede from in hps-mc jobs"""
+
+import shutil
+import re
+import os
+import logging
+
+from hpsmc.component import Component
+from ._util import Parameter
+from ._util import getBeamspotConstraintsFloatingOnly
+
+class PEDE(Component):
+    """! Run pede minimizer over input bin files for alignment
+    """
+
+    logger = logging.getLogger('hpsmc.tools.PEDE')
+
+    # each function is a "mask" in the sense that
+    #    it takes in a Parameter and should return True
+    #    if it should be floated and False if not
+    parameter_groups = {
+        'all' : lambda _p : True,
+        'allsensors' : lambda p : (p.mp_layer_id < 23),
+        'tu' : lambda p : (p.direction == 1 and p.trans_rot == 1),
+        'rw' : lambda p : (p.direction == 3 and p.trans_rot == 2)
```

Comment:
## Fundamental Groups
- direction
- operation
- individual vs structural
- back and front
- stereo and axial
- top and bottom
- hole and slot

## Logic
- add logic for converting a string into boolean operations so users can get the intersection of multiple groups